coral-spark/src/main/java/com/linkedin/coral/spark/transformers/ExtractUnionFunctionTransformer.java

```diff
@@ -56,6 +75,17 @@ protected SqlCall transform(SqlCall sqlCall) {
         createSqlOperator(COALESCE_STRUCT, CoalesceStructUtility.COALESCE_STRUCT_FUNCTION_RETURN_STRATEGY);
     if (operandList.size() == 1) {
       // one arg case: extract_union(field_name)
+      RelDataType operandType = deriveRelDatatype(sqlCall.operand(0));
+
+      if (containsSingleUnionType(operandType)) {
+        // Pass in schema string to keep track of the original Hive schema containing single uniontypes so coalesce_struct
```

Comment:
Expand the comment to explain why it is important for coalesce_struct to know this distinction.